samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -281,14 +290,22 @@ public void onContainerFailed(Throwable t) {
               LOGGER.warn("JobCoordinatorLatch was null. It is possible for some component to be waiting.");
             }
             containerException = t;
-            LOGGER.error(String.format("Container: %s failed with an exception. Stopping the stream processor: %s. Original exception:", container, processorId), containerException);
-            container = null;
+            SamzaContainer container = containerRef.getAndSet(null);
+            if (container != null) {
+              LOGGER.error(String.format("Container: %s failed with an exception. Stopping the stream processor: %s. Original exception:", container, processorId), containerException);
```

Comment:
I think it'd be useful to log the `processorId` and `containerException` even if container is null.